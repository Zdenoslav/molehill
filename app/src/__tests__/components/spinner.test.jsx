import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Spinner, {SpinnerSVG, Circle} from 'app/components/spinner';

describe('Spinner', () => {

  it('should render Spinner', () => {
    const wrapper = shallow(<Spinner />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render SpinnerSVG with margin', () => {
    const wrapper = renderer.create(<SpinnerSVG margined />).toJSON();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule('margin', '16px');
  });

  it('should render Circle with color', () => {
    const wrapper = renderer.create(<Circle color="red" />).toJSON();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule('stroke', 'red');
  });

});
