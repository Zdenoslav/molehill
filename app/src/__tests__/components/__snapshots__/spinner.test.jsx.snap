// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render Circle with color 1`] = `
.c0 {
  stroke: red;
  stroke-linecap: round;
  -webkit-animation: bBYgQX 1.5s ease-in-out infinite;
  animation: bBYgQX 1.5s ease-in-out infinite;
  stroke-width: 2px;
}

<circle
  className="c0"
  color="red"
/>
`;

exports[`Spinner should render Spinner 1`] = `
<Fragment>
  <styled.svg
    viewBox="0 0 24 24"
  >
    <styled.circle
      cx="12"
      cy="12 "
      fill="none"
      r="11"
    />
  </styled.svg>
</Fragment>
`;

exports[`Spinner should render SpinnerSVG with margin 1`] = `
.c0 {
  -webkit-animation: cZxgpV 2s linear infinite;
  animation: cZxgpV 2s linear infinite;
  z-index: 2;
  width: 20px;
  height: 20px;
  margin: 16px;
}

<svg
  className="c0"
/>
`;
